### ISSUE TITLE
Fine-tuning the code of llava, the loss decreases abnormally

### ISSUE THREAD

**Issue by binwang777** [OpenGVLab/InternVL#20]
I use the code you provided to train llava based on Intervit6B. According to the script you provided, the first stage of pretrain is running normally. But when using the fine-tuning script for training, I found strange loss transformations.

As shown in the picture:
![image](https://github.com/OpenGVLab/InternVL/assets/32870325/619d0e72-a841-4bcb-8a13-5b1ea58cf332)
I have not modified any llava code

I debugged EVA_clip_vit and found the problem. It happened in zero2. When replacing it with zero3, the loss was normal. However, Intervit6B has the following problems when using zero3:

![image](https://github.com/OpenGVLab/InternVL/assets/32870325/696100a8-9214-45a8-98f9-ffe790854c7c)

**Comment by czczup** [comment]
Hello, I'm sorry. I might have introduced some new bugs while organizing the code, and I am currently checking; I will fix it as soon as possible.

**Comment by binwang777** [comment]
Hi, regarding the problem of zero2 training loss being 0.0, I have fixed it here. After replacing the deepspeed version with 0.9.5, the loss converges normally. However, there are still problems with zero3 support, and the reason seems to be here:

![image](https://github.com/OpenGVLab/InternVL/assets/32870325/8c27ee49-1ce2-45c8-bbf2-4a27841d14d7)

**Comment by czczup** [comment]
> Hi, regarding the problem of zero2 training loss being 0.0, I have fixed it here. After replacing the deepspeed version with 0.9.5, the loss converges normally. However, there are still problems with zero3 support, and the reason seems to be here:
> 
> ![image](https://private-user-images.githubusercontent.com/32870325/293186899-8c27ee49-1ce2-45c8-bbf2-4a27841d14d7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM3NTkyMzcsIm5iZiI6MTcwMzc1ODkzNywicGF0aCI6Ii8zMjg3MDMyNS8yOTMxODY4OTktOGMyN2VlNDktMWNlMi00NWM4LWJiZjItNGEyNzg0MWQxNGQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI4VDEwMjIxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVhYjg3Y2VjMjBmNTA3ODE1OWRiMzJlMWQ3ZTYzZmM5Zjk0NGY3NGYzYTRjODk4YjZhY2MyY2I2NDQ4OTczZGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.dszRNXheFEFWmPh1YR_AEAloeXP4Ng3hjugQjhrF2no)

OK, I get it. Thanks for your feedback.

**Comment by czczup** [comment]
In the case of deepspeed zero3, it seems that resizing position encodings is not feasible. I am looking for a solution; I used to pre-resize and save them in advance, but I'm not too fond of doing it that way.

**Comment by czczup** [comment]
This bug should have been fixed. I have attempted both pre-training and fine-tuning, and the loss curves are normal. Additionally, the results from testing the newly trained models on the benchmarks are also normal.

**Comment by binwang777** [comment]
Hi, I have trained using the updated code. This time I only changed the dataset loading and weight loading, But my results are quite different from what you posted, is there still something wrong within the code.
![image](https://github.com/OpenGVLab/InternVL/assets/32870325/69238b97-94f2-4192-9c79-d1f9c352979e)
I suspect that flash_attn is not version 0.2.8 causing the problem, so I tested the weights you provided and the results are as follows:
![image](https://github.com/OpenGVLab/InternVL/assets/32870325/ebf9612a-2477-416d-ad5f-2a21e32c33ea)
The results are slightly fluctuating, but still reasonable.
